consensus/istanbul/backend/api.go

```diff
@@ -63,41 +67,65 @@ func (api *API) GetSnapshotAtHash(hash common.Hash) (*Snapshot, error) {
 	if header == nil {
 		return nil, errUnknownBlock
 	}
+
 	return api.istanbul.snapshot(api.chain, header.Number.Uint64(), header.Hash(), nil, false)
 }
 
 // GetValidators retrieves the list of authorized validators with the given block number.
 func (api *API) GetValidators(number *rpc.BlockNumber) ([]common.Address, error) {
-	snap, err := api.GetSnapshot(number)
+	previousBlockNumber := rpc.BlockNumber(number.Int64() - 1)
+
+	// get the snapshot of the previous block.
+	snap, err := api.GetSnapshot(&previousBlockNumber)
 	if err != nil {
-		return nil, err
+		logger.Error("Failed to get snapshot.", "hash", snap.Hash, "err", err)
+		return nil, errInternalError
```

Comment:
I think returning `err` is better because `GetSnapshot` is returns meaning error, too.
```suggestion
		return nil, err
```